src/buffer/out/ut_textbuffer/ReflowTests.cpp

```diff
@@ -0,0 +1,856 @@
+                        //   ^ cursor
+                    },
+                    { 1, 4 } },
+            },
+        },
+        TestCase{
+            L"SBCS, cursor off in space to far right of text (middle of buffer content), partially wrapped, aggressive shrink",
+            {
+                TestBuffer{
+                    { 6, 5 },
+                    {
+                        { L"ABCDEF", false },
+                        //       v cursor
+                        { L"$     ", false },
+                        //       ^ cursor
+                        { L"      ", false },
+                        { L"      ", true },
+                        { L"     Q", true },
+                    },
+                    { 5, 1 } // cursor in space far after $
+                },
+                TestBuffer{
+                    { 2, 5 }, // reduce width aggressively
+                    {
+                        { L"  ", true },
+                        { L"  ", true },
+                        { L"  ", true },
+                        { L" Q", true },
+                        //  v  cursor [BUG] cursor jumps to different place than fully wrapped case
+                        { L"  ", false },
+                        //  ^  cursor
+                    },
+                    { 0, 4 } },
+            },
+        },
+        TestCase{
+            // This triggers the cursor being walked forward w/ newlines to maintain
+            // distance from the last char in the buffer
+            L"SBCS, cursor at end of buffer, otherwise same as previous test",
+            {
+                TestBuffer{
+                    { 6, 5 },
+                    {
+                        { L"ABCDEF", false },
+                        { L"$     ", false },
+                        { L"     Q", true },
+                        { L"      ", true },
+                        //       v cursor
+                        { L"      ", true },
+                        //       ^ cursor
+                    },
+                    { 5, 4 } // cursor at end of buffer
+                },
+                TestBuffer{
+                    { 2, 5 }, // reduce width aggressively
+                    {
+                        { L"  ", true },
+                        { L"  ", true },
+                        { L" Q", true },
+                        { L"  ", false },
+                        //  v  cursor [BUG] cursor loses linear distance from Q; is this important?
+                        { L"  ", false },
+                        //  ^  cursor
+                    },
+                    { 0, 4 } },
+            },
+        },
+    };
+
+#pragma region TAEF hookup for the test case array above
+    struct ArrayIndexTaefAdapterRow : public Microsoft::WRL::RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::ClassicCom | Microsoft::WRL::InhibitFtmBase>, IDataRow>

```

Comment:
It'd be cool if this got pulled out so that other tests could take advantage of this. Alas, that's already on your radar and probably out of scope for this PR.